File: android/consentui/src/main/kotlin/com/consentui/ConsentResult.kt
package com.consentui

/**
 * Enum representation of consent result types for Java interoperability.
 * Use this enum in Java switch statements for easier handling.
 */
enum class ConsentResultType {
    ACCEPTED,
    DECLINED,
    NOT_REQUIRED
}

/**
 * Result of the consent request
 */
sealed class ConsentResult {
    /** The type of this result as an enum for Java interoperability */
    abstract val type: ConsentResultType

    /** User accepted the consent request */
    object Accepted : ConsentResult() {
        override val type: ConsentResultType = ConsentResultType.ACCEPTED
    }

    /** User declined the consent request */
    object Declined : ConsentResult() {
        override val type: ConsentResultType = ConsentResultType.DECLINED
    }

    /** Consent was not required (user not in EEA region) */
    object NotRequired : ConsentResult() {
        override val type: ConsentResultType = ConsentResultType.NOT_REQUIRED
    }

    companion object {
        /**
         * Creates a ConsentResult from the given type enum.
         * Useful for Java code that needs to convert from enum to sealed class.
         */
        @JvmStatic
        fun fromType(type: ConsentResultType): ConsentResult = when (type) {
            ConsentResultType.ACCEPTED -> Accepted
            ConsentResultType.DECLINED -> Declined
            ConsentResultType.NOT_REQUIRED -> NotRequired
        }
    }
}

/**
 * Callback interface for consent results
 */
interface ConsentCallback {
    fun onResult(result: ConsentResult)
}

File: ios/Sources/ConsentUI/ConsentResult.swift
import Foundation

/// Result of the consent request
public enum ConsentResult: Equatable {
    /// User accepted the consent request
    case accepted(attStatus: ATTStatus?)
    /// User declined the consent request
    /// ATT is still requested and status is returned
    case declined(attStatus: ATTStatus?)
    /// Consent was not required (user not in EEA region)
    /// ATT is still requested and status is returned
    case notRequired(attStatus: ATTStatus?)

    public static func == (lhs: ConsentResult, rhs: ConsentResult) -> Bool {
        switch (lhs, rhs) {
        case (.accepted(let lhsStatus), .accepted(let rhsStatus)):
            return lhsStatus == rhsStatus
        case (.declined(let lhsStatus), .declined(let rhsStatus)):
            return lhsStatus == rhsStatus
        case (.notRequired(let lhsStatus), .notRequired(let rhsStatus)):
            return lhsStatus == rhsStatus
        default:
            return false
        }
    }
}

/// App Tracking Transparency status
public enum ATTStatus: Equatable {
    case authorized
    case denied
    case restricted
    case notDetermined
}
